flytekit/common/workflow.py

```diff
@@ -286,6 +287,20 @@ def register(self, project, domain, name, version):
             self._id = old_id
             raise
 
+    @_exception_scopes.system_entry_point
+    def serialize(self):
```

Comment:
if this is going to be a required method on each registerable object, let's add it as an abstract method to the RegisterableEntity mixin.

Reply:
good point